utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,221 @@
+
+    string givenName = givenElement.data().trim();
+    string family = familyElement.data().trim();
+
+    if givenName != "" || family != "" {
+        return {given: [givenElement.data()], family: familyElement.data()};
+    } else {
+        log:printDebug("name fields not available");
+        return ();
+    }
+}
+
+# Map C-CDA code to FHIR CodeableConcept.
+#
+# + codingElement - C-CDA code element
+# + return - Return FHIR CodeableConcept
+public isolated function mapCcdaCodingtoFhirCodeableConcept(xml codingElement) returns r4:CodeableConcept? {
+    r4:Coding? mapCcdaCodingtoFhirCodeResult = mapCcdaCodingtoFhirCode(codingElement);
+    if mapCcdaCodingtoFhirCodeResult is r4:Coding {
+        return {
+            coding: [mapCcdaCodingtoFhirCodeResult]
+        };
+    } else {
+        log:printDebug("codeableConcept not available");
+        return ();
+    }
+}
+
+# Map C-CDA code to FHIR Coding.
+#
+# + codingElement - C-CDA code element
+# + return - Return FHIR Coding
+public isolated function mapCcdaCodingtoFhirCode(xml codingElement) returns r4:Coding? {
+    string|error? codeVal = codingElement.code;
+    string|error? systemVal = codingElement.codeSystem;
+
+    string? code = ();
+    if codeVal is string {
+        code = codeVal;
+    } else {
+        log:printDebug("codeVal is not available", codeVal);
+    }
+
+    string? system = ();
+    if systemVal is string {
+        system = systemVal;
+    } else {
+        log:printDebug("systemVal is not available", systemVal);
+    }
+
+    if code is string || system is string {
+        return {
+            code: code,
+            system: system
+        };
+    } else {
+        log:printDebug("coding fields not available");
+        return ();
+    }
+}
+
+# Map C-CDA dateTime to FHIR dateTime.
+#
+# + dateTimeElement - C-CDA dateTime element
+# + return - Return FHIR dateTime
+public isolated function mapCcdaDateTimeToFhirDateTime(xml dateTimeElement) returns r4:dateTime? {
+    string|error? dateTimeVal = dateTimeElement.value;
+    if dateTimeVal is string {
+        int length = dateTimeVal.length();
+        match (length) {
+            4 => {
+                return dateTimeVal;
+            }
+            6 => {
+                return string `${dateTimeVal.substring(0, 4)}-${dateTimeVal.substring(4, 6)}`;
+            }
+            8 => {
+                return string `${dateTimeVal.substring(0, 4)}-${dateTimeVal.substring(4, 6)}-${dateTimeVal.substring(6, 8)}`;
+            }
+            17 => {
+                return string `${dateTimeVal.substring(0, 4)}-${dateTimeVal.substring(4, 6)}-${dateTimeVal.substring(6, 8)}T${dateTimeVal.substring(9, 11)}:${dateTimeVal.substring(11, 15)}:${dateTimeVal.substring(15, 17)}`;
+            }
+            _ => {
+                log:printDebug("Invalid dateTime length");
+                return ();
+            }
+        }
+    }
+    log:printDebug("DateTime is not available", dateTimeVal);
+    return ();
```

Comment:
Do we need to explicitly return here?

Reply:
Please refer this discussion -> https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/pull/16#discussion_r1252777315